### PR TITLE
Initial version of logger

### PR DESCRIPTION
**Pull request by Taylorr82**

### CODE REVIEW

**Review by wagnerd97 — APPROVED** [review]

File: main/system.h [hunk 1]
```diff
@@ -23,6 +23,20 @@
...
+
+#if config_IS_ATHENA == 1
+#define IS_ATHENA
+#endif
+
+#if config_IS_FLATSAT == 1
+    #ifndef IS_ATHENA
+    #error If config_IS_FLATSAT is set to 1 IS_ATHENA must be defined
+    #endif
+#define IS_FLATSAT
```
Comment (ArashYzd) [inline main/system.h:37]: What are your thoughts on making IS_FLATSAT a predefined symbol so different hardware interfaces can also use? We then can get rid of all these EPS, UHF, etc. _IS_STUBBED macros.

**Review by ArashYzd — APPROVED** [review]

File: main/system.h (hunk 1 above)
Comment (Taylorr82) [reply main/system.h:37]: That is my intention. The way I implemented it in this file is to mirror the FreeRTOS config files. A predefined symbol may be better, we'd have to discuss

Comment (ArashYzd) [reply main/system.h:37]: Good! Yeah, I'll bring that up again once S-band and EPS tests are complete and integrated into the flatsat. This macro would be very useful, then.

File: ex2_system/source/logger/logger.c [hunk 2]
```diff
@@ -13,145 +13,218 @@
...
-    size_t tail;
-    size_t max; //of the buffer
-    bool full;
-};
+static bool fs_init; // true if filesystem initialized
 
-cbuf_handle_t circular_buf_init(uint8_t* buffer, size_t size)
-{
-    configASSERT(buffer && size);
+#define DEFAULT_INPUT_QUEUE_LEN 10
```
Comment (arrooney) [inline ex2_system/source/logger/logger.c:32]: put these above the globals

File: ex2_system/source/logger/logger.c [hunk 3]
```diff
@@ -13,145 +13,218 @@
...
+static TaskHandle_t my_handle;
 
-    cbuf->buffer = buffer;
-    cbuf->max = size;
-    circular_buf_reset(cbuf);
+const char logger_file[] = "VOL0:/syslog.log";
+uint32_t logger_file_handle = 0;
 
-    configASSERT(circular_buf_empty(cbuf));
+void test_logger_daemon(void *pvParameters);
```
Comment (arrooney) [inline ex2_system/source/logger/logger.c:41]: make it static

File: ex2_system/source/logger/logger.c [hunk 4]
```diff
@@ -13,145 +13,218 @@
...
+    } else {
+        task_name = pcTaskGetName(NULL);
+    }
 
-void circular_buf_free(cbuf_handle_t cbuf)
-{
-    configASSERT(cbuf);
-    free(cbuf);
-}
+    char buffer[PRINT_BUF_LEN+13] = {0};
```
Comment (arrooney) [inline ex2_system/source/logger/logger.c:67]: TASK_NAME_SIZE

File: ex2_system/source/logger/logger.c [hunk 5]
```diff
@@ -13,145 +13,218 @@
...
-}
-
-static void retreat_pointer(cbuf_handle_t cbuf)
-{
-    configASSERT(cbuf);
-
-    cbuf->full = FALSE;
-    cbuf->tail = (cbuf->tail + 1) % cbuf->max;
+    int32_t fd = red_open(logger_file, RED_O_RDWR | RED_O_APPEND);
+    if (fd == -1) {
```
Comment (arrooney) [inline ex2_system/source/logger/logger.c:134]: fd < 0

File: ex2_system/source/logger/logger.c [hunk 6]
```diff
@@ -13,145 +13,218 @@
...
+        int errno = red_errno;
+        fd = red_open(logger_file, RED_O_CREAT | RED_O_RDWR);
+        if (fd == -1) {
```
Comment (arrooney) [inline ex2_system/source/logger/logger.c:137]: fd < 0

File: ex2_system/source/logger/logger.c [hunk 7]
```diff
@@ -13,145 +13,218 @@
...
+ * 
+ * @details
+ * Closes syslog file and sets fs_init to false
+ */
+void stop_logger_fs() {
+    if (!fs_init) {
+        return;
+    }
+    red_close(logger_file_handle);
+    logger_file_handle = 0;
```
Comment (arrooney) [inline ex2_system/source/logger/logger.c:159]: Investigate this - do we need to protect this?

**Review by arrooney — COMMENTED** [review]
nice

File: ex2_system/source/logger/logger.c (hunk 7 above)
Comment (Taylorr82) [reply ex2_system/source/logger/logger.c:159]: I don't think so

**Pull request merged**